src/keygen/keyshare.rs

```diff
@@ -17,19 +22,23 @@ use zeroize::ZeroizeOnDrop;
 ///
 /// # 🔒 Storage requirements
 /// This type must be stored securely by the calling application.
-#[derive(Clone, ZeroizeOnDrop)]
+#[derive(Clone, ZeroizeOnDrop, Debug)]
 pub struct KeySharePrivate {
-    pub(crate) x: BigNumber, // in the range [1, q)
+    x: BigNumber, // in the range [1, q)
 }
 
-impl Debug for KeySharePrivate {
-    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        f.write_str("KeySharePrivate([redacted])")
+impl KeySharePrivate {
+    /// Get the private key share of the participant.
+    pub fn x(&self) -> BigNumber {
+        self.x.clone()
     }
-}
 
-impl KeySharePrivate {
-    // Computes the "raw" curve point corresponding to this private key.
+    /// Set x.
+    pub fn set_x(priv_share: BigNumber) -> KeySharePrivate {
+        KeySharePrivate { x: priv_share }
+    }
+
+    /// Computes the "raw" curve point corresponding to this private key.
     pub(crate) fn public_share(&self) -> Result<CurvePoint> {
```

Comment:
question: What do you think about changing this to return a `KeySharePublic` instead of the `CurvePoint`? 

Pros:
- I like the approach of not returning the raw point and forcing the caller to wrap it up correctly
- We could get rid of the `KeySharePublic::new` method. That method makes a public share from scratch, which doesn't make sense; a public key share should _always_ correspond to a private key share, otherwise it's just a random curve point.

Cons:
- We'd have to pass the correct `ParticipantId` as a parameter here. But right now the caller has to do that anyway, outside of this function, so it doesn't really make extra work, it just moves the work.

Reply:
That's fair, thank you.